### PR TITLE
LPD-15604 - Implement load more for search results

### PR DESCRIPTION
**Pull request by georgel-pop-lr**

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/js/select_layout/SelectLayoutTree.js [hunk 1]
```diff
@@ -448,41 +448,76 @@ export function SelectLayoutTree({
...
 	selection,
 }) {
+	const {findLayoutsURL, maxPageSize} = config;
 	const [results, setResults] = useState([]);
+	const [start, setStart] = useState(
+		Liferay.FeatureFlags['LPD-15607'] ? 0 : -1
+	);
+	const [loadMore, setLoadMore] = useState(false);
 	const [loading, setLoading] = useState(false);
+	const [prevFilter, setPrevFilter] = useState();
```
Comment (sandrodw3) [inline modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/js/select_layout/SelectLayoutTree.js:466]: This state and the `if` below are useless, right?

File: modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/js/select_layout/SelectLayoutTree.js [hunk 2]
```diff
@@ -448,41 +448,76 @@ export function SelectLayoutTree({
...
 
-	const onFindLayouts = useCallback((layouts) => {
-		setLoading(false);
+	const onFindLayouts = useCallback(
+		(layouts, hasMoreElements, start, prevLayouts) => {
+			setLoading(false);
 
-		setResults(layouts);
-	}, []);
+			setResults(prevLayouts.concat(layouts));
```
Comment (sandrodw3) [inline modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/js/select_layout/SelectLayoutTree.js:477]: The useState allows you to use the previous value 😄 you can read it [here](https://react.dev/reference/react/useState#setstate). So you can do:

`setResults((prevResults) => prevResults.concat(layouts))`

That way you can remove the `prevLayouts` param from all the code 🔥

**Review by sandrodw3 — COMMENTED** [review]
Hey @georgel-pop-lr this woks amazingly 😍 congratulations! Just left two suggestionss with some improvements and simplifications we can do.
Also remember to test everything without the FF to confirm it works as before this changes 🌈

File: modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/js/select_layout/SelectLayoutTree.js (hunk 1 above)
Comment (georgel-pop-lr) [reply modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/js/select_layout/SelectLayoutTree.js:466]: Yes, we can remove them

File: modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/js/select_layout/SelectLayoutTree.js (hunk 2 above)
Comment (georgel-pop-lr) [reply modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/js/select_layout/SelectLayoutTree.js:477]: Thanks for the hint, I will fix it.

**Comment by georgel-pop-lr** [comment]
> Hey @georgel-pop-lr this woks amazingly 😍 congratulations! Just left two suggestions with some improvements and simplifications we can do. Also remember to test everything without the FF to confirm it works as before this changes 🌈

Hi @sandrodw3

Thanks for your review, I did all the changes you propose, could you please check?

Regards.

**Comment by georgel-pop-lr** [comment]
ci:test:sf

**Comment by georgel-pop-lr** [comment]
Both unique failing tests are failing on master as well without my changes, so they are not related.

**Comment by georgel-pop-lr** [comment]
ci:forward:force

**Pull request closed without merging**